copy/copy.go

```diff
@@ -912,9 +919,9 @@ func (ic *imageCopier) copyLayers(ctx context.Context) error {
 		defer copyGroup.Wait()
 
 		for i, srcLayer := range srcInfos {
-			err = copySemaphore.Acquire(ctx, 1)
+			err = ic.c.copySemaphore.Acquire(ctx, 1)
 			if err != nil {
-				return errors.Wrapf(err, "Can't acquire semaphore")
+				return errors.Wrap(err, "copying layer")
```

Comment:
I should have moved it into a separate commit.  I found it useful to add context to the errors so that we can see where things broke without a stack trace.

Reply:
Oh, that’s my mistake — this change differentiates “failed acquiring the semaphore _while copying a layer vs. config_”, which is definitely useful and I completely failed to think about. The part I was focusing on was that this change loses the context that the failure is _acquiring semaphore vs. other failures during copy_, which is not actually necessary in the end I think (but it’s not trivial).

I wasn’t asking for a comment justifying the use of error wrapping just in this one place; I was asking for a comment justifying not including the “acquiring semaphore” context, maybe something like `// This can only fail with ctx.Err(), so no need to blame acquiring the semaphore

Reply:
Done :heavy_check_mark: